### ISSUE TITLE
Signs of overfitting?

### ISSUE THREAD

**Issue by HinLeung622** [Harry-Westwood/Y4-Project-InterNeuralStellar#13]
@grd349 
I am messing with a made up 2D mapping function that converts 2 inputs into 2 outputs, and trying to get a NN to learn that. To simulate mesa tracks, the input looks like this:
![image](https://user-images.githubusercontent.com/56083013/68358222-cc2f2380-010f-11ea-8170-d47efc38fb55.png)
And output:
![image](https://user-images.githubusercontent.com/56083013/68358226-d3563180-010f-11ea-9da9-2d479bce94fd.png)
I first did 4 hidden layers with 16 neurons each, and with 1000*10lines of data, and after 20k epoch, got this resulting plot:
![image](https://user-images.githubusercontent.com/56083013/68358286-fed91c00-010f-11ea-9bd0-c76a30180e3c.png)
blue plots the exact points used in training, but predicted by the NN, the background gray is the true outputs, and orange is the NN outputs for i have shifted the x1 inputs by +0.5*(distance between consecutive lines). Although the blue lines are not of a very accurate estimate of the true outputs, the top orange line does appear ~half interval distance above the top blue line, which seems correct. 
I thought it would do better if I expanded the architecture, so I made a NN that has also 4 layers but 64 neurons each, and this is its result after 20k epochs:
![image](https://user-images.githubusercontent.com/56083013/68358480-8d4d9d80-0110-11ea-89ae-d474f622fc39.png)
What is interesting is that, although this larger NN is able to get the blue lines more correct, the top orange line is now partially under the top blue line. and the overall evaluation loss score went up for a line that is within the input field (0 to 1 for both x1 and x2).

Question: is this all a sign of overfitting?
notebook is here: https://github.com/Harry-Westwood/Fourth-Year-Project/blob/master/Hin's_files/neural_network_test/TestNN_2D.ipynb
